src/main/java/frc/robot/subsystems/drive/Drivetrain.java

```diff
@@ -207,21 +256,56 @@ private void altDrive(Vector spdVec, Vector rotVec) {
     move(spdVec, rot);
   }
 
+  /**
+   * returns the current speed of the robot from it's reference frame
+   *
+   * @return the current speed of the robot from it's reference frame
+   */
+  public ChassisSpeeds getRobotRelativeSpeeds() {
+    return DriveConstants.kDriveKinematics.toChassisSpeeds(
+        new SwerveModuleState[] {
+          m_frontLeft.getState(),
+          m_frontRight.getState(),
+          m_rearLeft.getState(),
+          m_rearRight.getState()
+        });
+
+    // CCW rotation out of chassis frame
+    //   var rotated = new Translation2d(vxMetersPerSecond, vyMetersPerSecond).rotateBy(robotAngle);
+    //   return new ChassisSpeeds(rotated.getX(), rotated.getY(), omegaRadiansPerSecond);
+  }
+
+  /**
+   * applies smoothing to the turning input of altDrive
+   *
+   * @param stickAng the given angle of the driver turnig stick
+   * @return the commanded rotation based o the rotation input
+   */
   private double altTurnSmooth(double stickAng) {
     return Math.tanh(
             ((getGyroAngle().in(Units.Radians) + stickAng + Math.PI) % (2 * Math.PI) - Math.PI)
                 / DriveConfig.altTurnSmoothing)
         * DriveConfig.kMaxAngularSpeed;
   }
 
+  /**
+   * returns the current position of the robot on the field
+   *
+   * @return the current position of the robot on the field
+   */
+  private Pose2d getPose() {
+    Pose2d pose = m_odometry.getPoseMeters();
+    return (pose);
+  }
+
   /**
    * moves the drivetrain using the given values
    *
    * @param xSpeed the proportion of the robot's max velocity to move in the x direction
    * @param ySpeed the proportion of the robot's max velocity to move in the y direction
    * @param rot the angular velocity to rotate the drivetrain in radians/s
    */
-  private void move(Vector spdVec, double rot) {
+  public void move(Vector spdVec, double rot) {
```

Comment:
Why change from private to public?

Reply:
it's as a convenience for commands where directly giving a target movement and rotation is often going to be more convenient than giving fake controller stick vectors